File: src/main/java/com/eager/questioncloud/workspace/WorkSpaceQuestionService.java
package com.eager.questioncloud.workspace;

import com.eager.questioncloud.question.Question;
import com.eager.questioncloud.question.QuestionContent;
import com.eager.questioncloud.question.QuestionDto.QuestionInformationForWorkSpace;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class WorkSpaceQuestionService {
    private final WorkSpaceQuestionRegister workSpaceQuestionRegister;
    private final WorkSpaceQuestionUpdater workSpaceQuestionUpdater;
    private final WorkSpaceQuestionRemover workSpaceQuestionRemover;
    private final WorkSpaceQuestionReader workSpaceQuestionReader;

    public int count(Long creatorId) {
        return workSpaceQuestionReader.count(creatorId);
    }

    public List<QuestionInformationForWorkSpace> getQuestions(Long creatorId, Pageable pageable) {
        return workSpaceQuestionReader.getQuestions(creatorId, pageable);
    }

    public QuestionContent getQuestionContent(Long creatorId, Long questionId) {
        return workSpaceQuestionReader.getQuestionContent(creatorId, questionId);
    }

    public Question register(Long creatorId, QuestionContent questionContent) {
        return workSpaceQuestionRegister.register(creatorId, questionContent);
    }

    public void modify(Long creatorId, Long questionId, QuestionContent questionContent) {
        workSpaceQuestionUpdater.modify(creatorId, questionId, questionContent);
    }

    public void delete(Long creatorId, Long questionId) {
        workSpaceQuestionRemover.remove(creatorId, questionId);
    }
}

File: qc-core/qc-workspace/qc-workspace-api/src/main/kotlin/com/eager/questioncloud/workspace/service/WorkspaceQuestionService.kt
package com.eager.questioncloud.workspace.service

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.workspace.command.DeleteQuestionCommand
import com.eager.questioncloud.workspace.command.ModifyQuestionCommand
import com.eager.questioncloud.workspace.command.RegisterQuestionCommand
import com.eager.questioncloud.workspace.dto.CreatorQuestionInformation
import com.eager.questioncloud.workspace.dto.MyQuestionContent
import com.eager.questioncloud.workspace.implement.WorkspaceQuestionReader
import com.eager.questioncloud.workspace.implement.WorkspaceQuestionRegister
import com.eager.questioncloud.workspace.implement.WorkspaceQuestionRemover
import com.eager.questioncloud.workspace.implement.WorkspaceQuestionUpdater
import org.springframework.stereotype.Component

@Component
class WorkspaceQuestionService(
    private val workspaceQuestionReader: WorkspaceQuestionReader,
    private val workspaceQuestionRegister: WorkspaceQuestionRegister,
    private val workspaceQuestionUpdater: WorkspaceQuestionUpdater,
    private val workspaceQuestionRemover: WorkspaceQuestionRemover,
) {
    fun getMyQuestions(creatorId: Long, pagingInformation: PagingInformation): List<CreatorQuestionInformation> {
        return workspaceQuestionReader.getMyQuestions(creatorId, pagingInformation)
    }
    
    fun countMyQuestions(creatorId: Long): Int {
        return workspaceQuestionReader.countMyQuestions(creatorId)
    }
    
    fun getMyQuestionContent(creatorId: Long, questionId: Long): MyQuestionContent {
        return workspaceQuestionReader.getMyQuestionContent(creatorId, questionId)
    }
    
    fun registerQuestion(command: RegisterQuestionCommand) {
        workspaceQuestionRegister.registerQuestion(command)
    }
    
    fun modifyQuestion(command: ModifyQuestionCommand) {
        workspaceQuestionUpdater.modifyQuestion(command)
    }
    
    fun deleteQuestion(command: DeleteQuestionCommand) {
        workspaceQuestionRemover.deleteQuestion(command)
    }
}
